CMakeLists.txt

```
@@ -4248,6 +4267,13 @@ if(NOT WIN32)
   )
 endif()
 
+if(BUILD_BENCH)
+  if(NOT BUILD_TESTING)
```

For a bit less headaches: (need to be moved before all other checks.
```suggestion
  if(NOT DEFINED BUILD_TESTING)
    set(BUILD_TESTING ON)
  endif()
  if(NOT BUILD_TESTING)   
```